### PR TITLE
ENH: change send -> __call__ for partials

### PR DESCRIPTION
**Pull request by adler-j**
Solves issue #213 and some other style issues with partials:

* Changed `partial.send(x)` to `partial(x)`
* Add `PrintTimePartial` (cudos do @moosmann for code)

### CODE REVIEW

File: odl/solvers/util/partial.py [hunk 1]
```diff
@@ -37,7 +38,7 @@ class Partial(with_metaclass(ABCMeta, object)):
     """Abstract base class for sending partial results of iterations."""
 
     @abstractmethod
-    def send(self, result):
+    def __call__(self, result):
         """Send the result to the partial object."""
```
Comment (kohr-h) [inline odl/solvers/util/partial.py]: Somewhat twisted language, given that "send" does no longer appear in the name.

File: odl/solvers/util/partial.py [hunk 2]
```diff
@@ -108,15 +130,30 @@ class PrintIterationPartial(Partial):
...
 
-    def send(self, _):
+    def __call__(self, _):
         """Print the current iteration."""
-        print("iter = {}".format(self.iter))
+        print("{} = {}".format(self.text, self.iter))
         self.iter += 1
 
 
+class PrintTimePartial(Partial):
```
Comment (kohr-h) [inline odl/solvers/util/partial.py]: I'd call it `PrintTimeDiffPartial`, only time suggests absolute time.

File: odl/solvers/util/partial.py [hunk 3]
```diff
@@ -108,15 +130,30 @@ class PrintIterationPartial(Partial):
...
+
+    """Print the time elapsed to the previous iteration."""
```
Comment (kohr-h) [inline odl/solvers/util/partial.py]: since

File: odl/solvers/util/partial.py [hunk 4]
```diff
@@ -135,13 +172,31 @@ class ShowPartial(Partial):
...
+
+        parameters are passed through to the vectors show method. Additional
+        parameters include
+
+        Parameters
+        ----------
+        *args, **kwargs
+            passed ax ``x.show(*args, **kwargs)``
+        display_step : positive int
+            Plot every n:th image, default: 1
```
Comment (kohr-h) [inline odl/solvers/util/partial.py]: Swedish ordinal :-) Perhaps write "Number of iterations between plots. Default: 1" (default always as a new "sentence")

File: odl/solvers/util/partial.py [hunk 5]
```diff
@@ -53,7 +54,8 @@ def __and__(self, other):
         Returns
         -------
         result : `Partial`
-            A partial which's `send` method calls both constituends partials.
+            A partial which's `__call__` method calls both constituends
```
Comment (kohr-h) [inline odl/solvers/util/partial.py]: whose

File: odl/solvers/util/partial.py [hunk 6]
```diff
@@ -99,7 +121,7 @@ class ForEachPartial(Partial):
     def __init__(self, function):
         self.function = function
 
-    def send(self, result):
+    def __call__(self, result):
         """Applies function to result."""
```
Comment (kohr-h) [inline odl/solvers/util/partial.py]: Apply

File: odl/solvers/util/partial.py [hunk 7]
```diff
@@ -108,15 +130,30 @@ class PrintIterationPartial(Partial):
...
-    def __init__(self):
+    def __init__(self, text=None):
+        self.text = text if text is not None else 'iter'
         self.iter = 0
 
-    def send(self, _):
+    def __call__(self, _):
         """Print the current iteration."""
-        print("iter = {}".format(self.iter))
+        print("{} = {}".format(self.text, self.iter))
```
Comment (kohr-h) [inline odl/solvers/util/partial.py]: I'd leave out the `=` in the custom format and add use `iter = ` as a default.

File: odl/solvers/util/partial.py [hunk 8]
```diff
@@ -82,15 +90,29 @@ class StorePartial(Partial):
...
 
-    def send(self, result):
+    def __call__(self, result):
         """Append result to results list."""
-        self.results.append(result.copy())
+        self._results.append(result.copy())
 
     def __iter__(self):
+        """Allow iteration over the results"""
         return self.results.__iter__()
```
Comment (kohr-h) [inline odl/solvers/util/partial.py]: `iter(self.results)`

**Comment by kohr-h** [comment]
Nice to see docstrings appearing. Some are still missing, though, at places where they could be useful, e.g. in `__init__` of the timer partial. Fix that and merge.

File: odl/solvers/util/partial.py (hunk 2 above)
Comment (adler-j) [inline odl/solvers/util/partial.py]: Went for `PrintTimingPartial` which impies that it does a timing, thought `PrintTimeDiffPartial` was somewhat long?

File: odl/solvers/util/partial.py (hunk 4 above)
Comment (adler-j) [inline odl/solvers/util/partial.py]: :+1:

**Comment by adler-j** [comment]
Fixed all comments, will merge after Travis is happy.

File: odl/solvers/util/partial.py (hunk 2 above)
Comment (kohr-h) [inline odl/solvers/util/partial.py]: Saving 2 letters.. but ok

**Pull request merged**